File: parts/p011/src/main/java/org/rapidpm/event/mutationtesting/p011/first/Service.java
package org.rapidpm.event.mutationtesting.p011.first;

/**
 * Create the best test coverage you can.
 *
 * How many tests you will need?
 * What are the "good" tests?
 */
public class Service {
	/**
	 * A non-sense add method
	 * @param a
	 * @param b
	 * @return a usefull result
	 */
	public int add(int a, int b) {
		if (a < 2) {
			return (a + b) * -1;
		} else {
			return a + b;
		}
	}
}

File: parts/p002/src/main/kotlin/junit/org/rapidpm/event/mutationtesting/p002/Service.kt
package junit.org.rapidpm.event.mutationtesting.p002

/**
 * Create the best test coverage you can.
 * How many tests you will need?
 * What are the "good" tests?
 */
class Service {

    fun add(a: Int, b: Int): Int =
            when {
                a < 2 -> (a + b) * -1
                b > 5 && a < 2 -> b * (a + 1)
                b > 7 && a > 4 -> (b+3) * (-a)
                else -> a + b
            }
}